akka-http-core/src/main/java/akka/http/javadsl/model/MediaTypes.java

```diff
@@ -187,6 +187,119 @@ private MediaTypes() { }
     public static final MediaType.Binary VIDEO_X_SGI_MOVIE = akka.http.scaladsl.model.MediaTypes.video$divx$minussgi$minusmovie();
     public static final MediaType.Binary VIDEO_WEBM = akka.http.scaladsl.model.MediaTypes.video$divwebm();
 
+    public static MediaType.Binary applicationBinary(String subType, boolean compressible, String... fileExtensions) {
+        akka.http.scaladsl.model.MediaType.Compressibility comp = compressible ?
+                akka.http.scaladsl.model.MediaType.Compressible$.MODULE$ : akka.http.scaladsl.model.MediaType.NotCompressible$.MODULE$;
+
+        scala.collection.immutable.List<String> fileEx =
+                scala.collection.JavaConversions.asScalaBuffer(java.util.Arrays.asList(fileExtensions)).toList();
+
+        return akka.http.scaladsl.model.MediaType.applicationBinary(subType, comp, fileEx);
+    }
+
+    public static MediaType.WithFixedCharset applicationWithFixedCharset(String subType, HttpCharset charset, String... fileExtensions) {
+        akka.http.scaladsl.model.HttpCharset cs = (akka.http.scaladsl.model.HttpCharset) charset;
```

Comment:
It will be casts like this indeed. It's "good" because we don't need wrappers (though it's "bad" because casting...).
In scala we have a safe utility in `JavaMapping` to make those casts safetly, I'm not sure if we can use it in java code